### PR TITLE
Support quoted headers

### PR DESCRIPTION
**Pull request by charleso**
/cc @markhibberd @jystic @EduardSergeev 

This is patched on the version of htoml we're currently using (not the latest)

### CODE REVIEW

**Comment by EduardSergeev** [comment]
Worth extending [the test](https://github.com/ambiata/htoml/blob/master/test/Text/Toml/Parser/Spec.hs#L134-L152)? 

Right from the [spec](https://github.com/toml-lang/toml#table) if I am not mistaken?:
```
      testParser headerValue "[dog.\"tater.man\"]" ["dog", "\"tater.man\""]
```

**Comment by charleso** [comment]
Have added the test -look ok?

**Comment by EduardSergeev** [comment]
Assuming it's passing :+1:

File: test/Text/Toml/Parser/Spec.hs [hunk 1]
```diff
@@ -151,6 +151,9 @@ tomlParserSpec = testSpec "Parser Hspec suite" $ do
     it "should not parse an empty middle name" $
       testParserFails headerValue  "first..second"
 
+    if "should parse tables with quotes" $
```
Comment (EduardSergeev) [inline test/Text/Toml/Parser/Spec.hs]: /s/if/it/g
And the test is actually failing, I believe

Comment (EduardSergeev) [inline test/Text/Toml/Parser/Spec.hs]: Latest version of test passes, sorry for the false alarm.

**Comment by EduardSergeev** [comment]
:+1:

**Pull request merged**